underfs/cosn/src/main/java/alluxio/underfs/cosn/CosNUnderFileSystemFactory.java

```diff
@@ -40,6 +42,17 @@ public boolean supportsPath(String path) {
 
   @Override
   public boolean supportsPath(String path, UnderFileSystemConfiguration conf) {
-    return supportsPath(path);
+    if (supportsPath(path)) {
+      if (!conf.isSetByUser(PropertyKey.UNDERFS_VERSION)
+          || conf.get(PropertyKey.UNDERFS_VERSION).equals(getVersion())) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
```suggestion
    if (!supportsPath(path)) {
      return false;
    }
    // if the user has explicitly specified a UFS version, do a strict match
    if (conf.isSetByUser(PropertyKey.UNDERFS_VERSION)
        && !conf.get(PropertyKey.UNDERFS_VERSION).equals(getVersion())) {
      return false;
    }
    // otherwise, assume the version we ship with Alluxio supports it.
    return true;
```

Reply:
done